File: Frontend/src/app/@models/meta-model/api-result.ts
export interface ApiResultStatus {
  code: number;
  status(): ResultStatus;
}

export interface ApiResult extends ApiResultStatus {
  code: number;
  message?: string;
  developer?: any;
  method?: MessageMethod;
}

export interface ApiResultData<Model> extends ApiResult {
  data?: Model;
}

export enum MessageMethod {
  toaster,
  dialog,
  alert,
  console,
}

export enum ShowMessageCase {
  silence,
  errors,
  success,
}

export enum ResultStatus {
  info = 1,
  success,
  warning,
  error,
  serverFatal,
  uiFatal,
}

File: Backend/Infrastructure/Controller/ApiResult.cs
using System.Reflection;

namespace PhotonBypass.Infra.Controller;

public class ApiResult
{
    public short Code { get; set; }

    public string? Message { get; set; }

    public string? Developer { get; set; }

    public MessageMethod? MessageMethod { get; set; }

    public static ApiResult Success(string message)
    {
        return new ApiResult
        {
            Code = 200,
            Message = message,
        };
    }
}

public class ApiResult<Model> : ApiResult
{
    public Model? Data { get; set; }

    public static ApiResult<Model> Success(Model model)
    {
        return new ApiResult<Model>
        {
            Code = 200,
            Data = model,
        };
    }
}